packages/auth/amplify_auth_cognito_dart/lib/src/sdk/sdk_exception.dart

```diff
@@ -154,24 +144,46 @@ class InvalidParameterException extends core.AmplifyException
   String get runtimeTypeName => 'InvalidParameterException';
 }
 
-const _lamdaDeprecationNotes = '''
-This type wraps the following exceptions and should be replaced by one of them:
-InvalidLambdaResponseException
-UnexpectedLambdaException
-UserLambdaValidationException''';
-
 /// {@template amplify_auth_cognito_dart.sdk.lambda_exception}
 /// Exception thrown when an error from the AWS Lambda service is encountered.
 /// {@endtemplate}
-@Deprecated(_lamdaDeprecationNotes)
 class LambdaException extends core.AmplifyException with core.AWSDebuggable {
   /// {@macro amplify_auth_cognito_dart.sdk.lambda_exception}
-  @Deprecated(_lamdaDeprecationNotes)
-  const LambdaException(
+  factory LambdaException(String message, {String? underlyingException}) {
+    final match = _errorRegex.firstMatch(message);
+    final lambdaName = match?.group(1);
+    final parsedMessage = match?.group(2);
+    if (parsedMessage != null) {
+      message = parsedMessage;
+    }
+    return LambdaException._(
+      message,
+      lambdaName: lambdaName,
+      underlyingException: underlyingException,
+    );
+  }
+
+  const LambdaException._(
     super.message, {
+    this.lambdaName,
     super.underlyingException,
   });
 
+  /// Whether [exception] originated in a user Lambda.
+  static bool isLambdaException(String exception) =>
```

Comment:
I think you're correct that this is really the only way to infer that these were originating in Lambda, but it seems like kind of a fragile approach in that a change to a string sent back from Cognito in a non-Lambda exception using this patter would result in an incorrect inference. Would it be better to call these UnknownExceptions and be correct-but-vague, rather than call them LambdaExceptions and be more-precise-but-possibly-wrong?

Reply:
Alrighty  - I can definitely see that avoiding the UserNotFoundException is kind of imperative.